### ISSUE TITLE
Crash opening NSF file with exception "String conversion error"

### ISSUE THREAD

**Issue by velipso** [BleuBleu/FamiStudio#163]
Opening a random NSF file I found on the internet causes FamiStudio to crash.  Stack trace and file below.

[Jetsons, The - Cogswell's Caper! (J).nsf.zip](https://github.com/BleuBleu/FamiStudio/files/9620905/Jetsons.The.-.Cogswell.s.Caper.J.nsf.zip)

```
Unhandled Exception:
System.ExecutionEngineException: String conversion error: Illegal byte sequence encounted in the input.
  at (wrapper managed-to-native) System.Runtime.InteropServices.Marshal.PtrToStringAnsi(intptr)
  at FamiStudio.NsfFile.Load (System.String filename, System.Int32 songIndex, System.Int32 duration, System.Int32 patternLength, System.Int32 startFrame, System.Boolean removeIntroSilence, System.Boolean reverseDpcm, System.Boolean preserveDpcmPad) [0x000c0] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.NsfImportDialog+<>c__DisplayClass4_0.<ShowDialog>b__0 (FamiStudio.DialogResult r) [0x000c1] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Dialog.ShowDialogAsync (System.Action`1[T] cb) [0x00007] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.NsfImportDialog.ShowDialog (FamiStudio.FamiStudioWindow parent) [0x0001c] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.OpenProjectFile (System.String filename, System.Boolean allowComplexFormats) [0x000cc] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.OpenProjectInternal (System.String filename) [0x00010] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio+<>c__DisplayClass239_0.<OpenProject>b__0 (System.String f) [0x0000b] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio+<>c__DisplayClass239_0.<OpenProject>b__1 () [0x00035] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.TrySaveProjectAsync (System.Action callback) [0x0004c] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.OpenProject (System.String filename) [0x00026] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Toolbar.OnOpen (System.Int32 x, System.Int32 y) [0x00006] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Toolbar.OnMouseDown (FamiStudio.MouseEventArgs e) [0x00063] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Control.MouseDown (FamiStudio.MouseEventArgs e) [0x00000] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudioWindow.MouseButtonCallback (System.IntPtr window, System.Int32 button, System.Int32 action, System.Int32 mods) [0x00123] in <4059daf93a3e4b00abab38a450d40298>:0 
  at (wrapper native-to-managed) FamiStudio.FamiStudioWindow.MouseButtonCallback(intptr,int,int,int)
  at (wrapper managed-to-native) System.Object.wrapper_native_0x102f3c780()
  at GLFWDotNet.GLFW.glfwPollEvents () [0x00000] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudioWindow.ProcessEvents () [0x00000] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudioWindow.RunIteration (System.Boolean allowSleep) [0x00006] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudioWindow.Run () [0x00002] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.Run (System.String[] args) [0x0002f] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Program.Main (System.String[] args) [0x00028] in <4059daf93a3e4b00abab38a450d40298>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ExecutionEngineException: String conversion error: Illegal byte sequence encounted in the input.
  at (wrapper managed-to-native) System.Runtime.InteropServices.Marshal.PtrToStringAnsi(intptr)
  at FamiStudio.NsfFile.Load (System.String filename, System.Int32 songIndex, System.Int32 duration, System.Int32 patternLength, System.Int32 startFrame, System.Boolean removeIntroSilence, System.Boolean reverseDpcm, System.Boolean preserveDpcmPad) [0x000c0] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.NsfImportDialog+<>c__DisplayClass4_0.<ShowDialog>b__0 (FamiStudio.DialogResult r) [0x000c1] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Dialog.ShowDialogAsync (System.Action`1[T] cb) [0x00007] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.NsfImportDialog.ShowDialog (FamiStudio.FamiStudioWindow parent) [0x0001c] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.OpenProjectFile (System.String filename, System.Boolean allowComplexFormats) [0x000cc] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.OpenProjectInternal (System.String filename) [0x00010] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio+<>c__DisplayClass239_0.<OpenProject>b__0 (System.String f) [0x0000b] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio+<>c__DisplayClass239_0.<OpenProject>b__1 () [0x00035] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.TrySaveProjectAsync (System.Action callback) [0x0004c] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.OpenProject (System.String filename) [0x00026] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Toolbar.OnOpen (System.Int32 x, System.Int32 y) [0x00006] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Toolbar.OnMouseDown (FamiStudio.MouseEventArgs e) [0x00063] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Control.MouseDown (FamiStudio.MouseEventArgs e) [0x00000] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudioWindow.MouseButtonCallback (System.IntPtr window, System.Int32 button, System.Int32 action, System.Int32 mods) [0x00123] in <4059daf93a3e4b00abab38a450d40298>:0 
  at (wrapper native-to-managed) FamiStudio.FamiStudioWindow.MouseButtonCallback(intptr,int,int,int)
  at (wrapper managed-to-native) System.Object.wrapper_native_0x102f3c780()
  at GLFWDotNet.GLFW.glfwPollEvents () [0x00000] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudioWindow.ProcessEvents () [0x00000] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudioWindow.RunIteration (System.Boolean allowSleep) [0x00006] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudioWindow.Run () [0x00002] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.FamiStudio.Run (System.String[] args) [0x0002f] in <4059daf93a3e4b00abab38a450d40298>:0 
  at FamiStudio.Program.Main (System.String[] args) [0x00028] in <4059daf93a3e4b00abab38a450d40298>:0 ```

**Comment by BleuBleu** [comment]
Oh thanks!! Will take a look in the next couple of days.

-Mat

**Comment by BleuBleu** [comment]
Fixed 4.0.2.
